### DISCUSSION TITLE
Convert a string to an IPv6 prefix

### DISCUSSION THREAD

**Discussion by Mikeadaneg** [openthread/openthread#9014] [Q&A] [answered]
I noticed that the OpenThread library provides a function called otIp6PrefixToString in ip6.h to convert an IPv6 prefix to a string, but there doesn't seem to be a function to do the opposite (i.e., convert a string to an IPv6 prefix). Is there any plan to add such a function in the future?

I also found a C++ function called ParseAsIp6Prefix in parse_cmdline.c that can parse an IPv6 prefix from a string. Is there a code example available on how to wrap this function into a C-compatible interface that can be called from a C program?

**Comment by jwhui** [comment]
See [otIp6AddressFromString](https://openthread.io/reference/group/api-ip6#otip6addressfromstring)

**Reply by Mikeadaneg** [reply, marked as answer]
Thanks!  I was looking for otStringToIp6Prefix.

**Comment by abtink** [comment]
@Mikeadaneg I added `otIp6PrefixFromString()` in the PR:
- https://github.com/openthread/openthread/pull/9016.
